### PR TITLE
Fix issue #1396 (invalid header comparison)

### PR DESCRIPTION
**Pull request by sstone**

### CODE REVIEW

File: eclair-core/src/test/scala/fr/acinq/eclair/blockchain/electrum/ElectrumWalletSimulatedClientSpec.scala [hunk 1]
```diff
@@ -89,30 +101,24 @@ class ElectrumWalletSimulatedClientSpec extends TestKit(ActorSystem("test")) wit
...
-
   test("tell wallet is ready when a new block comes in, even if nothing else has changed") {
+    val f = fixture
+    import f._
+
     val last = wallet.stateData.blockchain.tip
     assert(last.header == headers.last)
     val header = makeHeader(last.header)
-    headers = headers :+ header
+    //headers = headers :+ header
```
Comment (t-bast) [inline eclair-core/src/test/scala/fr/acinq/eclair/blockchain/electrum/ElectrumWalletSimulatedClientSpec.scala:111]: Hmm, shouldn't this be uncommented?

**Comment by t-bast** [comment]
Don't we also need to apply this on master?

**Comment by pm47** [comment]
> Don't we also need to apply this on master?

Definitely!

**Comment by sstone** [comment]
The faulty header comparison as already been fixed on master (as part of the switch to scala 2.13). I'm not even sure we should keep the Electrum code in master now ?

File: eclair-core/src/test/scala/fr/acinq/eclair/blockchain/electrum/ElectrumWalletSimulatedClientSpec.scala (hunk 1 above)
Comment (sstone) [reply eclair-core/src/test/scala/fr/acinq/eclair/blockchain/electrum/ElectrumWalletSimulatedClientSpec.scala]: This was useless, I've removed it now

**Comment by t-bast** [comment]
> I'm not even sure we should keep the Electrum code in master now ?

Good point, in the long term we probably want to remove it (the `Setup` code will be simpler, the configuration too, and we'll be making the codebase smaller) but I think for now it's safer to keep it until we delete the android branch?

**Pull request merged**